### PR TITLE
expression: implement vectorized evaluation for builtinCastTimeAsTimeSig

### PR DESCRIPTION
**Pull request by ekalinin**
PCP #12101

### What problem does this PR solve? 

This PR implements vectorized `makebuiltinCastTimeAsTimeSig`.

### What is changed and how it works?

```sh
➜ make vectorized-bench VB_FILE=Cast VB_FUNC=builtinCastTimeAsTimeSig
cd ./expression && \
	go test -v -benchmem \
		-bench=BenchmarkVectorizedBuiltinCastFunc \
		-run=BenchmarkVectorizedBuiltinCastFunc \
		-args "builtinCastTimeAsTimeSig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinCastFunc/builtinCastTimeAsTimeSig-VecBuiltinFunc-12         	   26919	     44802 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastTimeAsTimeSig-NonVecBuiltinFunc-12      	   17143	     70504 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastTimeAsTimeSig-VecBuiltinFunc#01-12      	   17361	     69502 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastTimeAsTimeSig-NonVecBuiltinFunc#01-12   	   12207	     99565 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	7.719s
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_cast_vec.go [hunk 1]
```diff
@@ -1053,11 +1053,38 @@ func (b *builtinCastTimeAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
...
+		}
+		res, err := times[i].Convert(stmt, b.tp.Tp)
+		if err != nil {
+			if err = handleInvalidTimeError(b.ctx, err); err != nil {
+				return err
+			}
+			result.SetNull(i, true)
+			continue
+		}
+		tm, err := res.RoundFrac(stmt, fsp)
```
Comment (wshwsh12) [inline expression/builtin_cast_vec.go]: If err != nil , return err ?

File: expression/builtin_cast_vec.go [hunk 2]
```diff
@@ -1053,11 +1053,38 @@ func (b *builtinCastTimeAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
...
+		times[i] = tm
+		if b.tp.Tp == mysql.TypeDate {
+			// Truncate hh:mm:ss part if the type is Date.
+			times[i].Time = types.FromDate(tm.Time.Year(), tm.Time.Month(), tm.Time.Day(), 0, 0, 0, 0)
+		}
```
Comment (wshwsh12) [inline expression/builtin_cast_vec.go]: add a line.  `times[i].Type = b.tp.Tp`

File: expression/builtin_cast_vec.go (hunk 1 above)
Comment (ekalinin) [reply expression/builtin_cast_vec.go]: Fxied.

File: expression/builtin_cast_vec.go (hunk 2 above)
Comment (ekalinin) [reply expression/builtin_cast_vec.go]: Fixed

**Pull request merged**